### ISSUE TITLE
Hub(auth) is not firing, when logged on via refresh/existing cookie log on (working in v5 not working in v6)

### ISSUE THREAD

**Issue by myonara** [aws-amplify/amplify-js#12739]
### Before creating a new issue, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-ui/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-ui/discussions).
- [X] I have tried disabling all browser extensions or using a different browser
- [X] I have tried deleting the node_modules folder and reinstalling my dependencies
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md#bug-reports).

### On which framework/platform are you having an issue?

Angular

### Which UI component?

Authenticator

### How is your app built?

angular/amplify

### What browsers are you seeing the problem on?

Chrome, Firefox

### Which region are you seeing the problem in?

eu-central-1 / localhost execution

### Please describe your bug.

In version 6 I'm getting events ob the hub-Channel "auth" for signedin and signedup, but getting no hub event, when I refresh the browser and have a valid auth cookie. In version 5 I'm getting signin events also, when I'm refreshing the browser with an auth cookie. Side question: Also I used Auth.currentSession in V5, which is missing in V6, is there a replacement for this?

### What's the expected behaviour?

The hub-channel auth can get also re signed in events.

### Help us reproduce the bug!

Just do a default project with auth in V6 and listen to the Hub-Auth-Channel during the steps of singing up/signing in and refresh browser after logged in.

### Code Snippet

```typescript
// Put your code below this line.
    this.hubListenerCancelToken = Hub.listen('auth', async (msg: any) => {
      console.log('hub-1', msg.payload.event);
    }
```

### Console log output

```
// getting signin,signup in V5 also for refreshing page with an auth cookie set.
// getting signedin,sgnedup in V6 but not signedin during reloading the page with an auth cookie!
```

**Comment by cwomack** [comment]
Hello, @myonara and sorry for the delayed response on this issue.  The equivalent to `Auth.currentSession()` from v5 would be the `fetchAuthSession()` potentially for what you're looking to use ([docs here](https://docs.amplify.aws/react/build-a-backend/auth/manage-user-session/#retrieve-a-user-session)).  There's also an updated section on the Auth events from Hub [here](https://docs.amplify.aws/react/build-a-backend/auth/auth-events/) if you'd like more details on how to use them in v6. 

Let me know if that helps clear things up or if there's a specific use case or context we can help with further!

**Comment by nadetastic** [comment]
Hi @myonara following up here - let us know if the comment from @cwomack above addressed your issue. Thanks!

**Comment by myonara** [comment]
Sorry, just seen it now, thank you for the reminder. I will dig into the docs and code soon to clarify.

**Comment by myonara** [comment]
So I have tested the hints, @cwomack has given. Now my code is working again. Here are my observations:
* There is no Hub Auth-Event "refreshToken" without using fetchAuthSession()
* Do not call fetchAuthSession({ forceRefresh: true }) within the the handling of "refreshToken", that produces an endless loop. The event is fired before the the fetchAuthSession() is returning values in the client.
* authToken and idToken is delivered with fetchAuthSession({ forceRefresh: true }), both have the identical sub field
* the idToken is marked as optional in the definition, but it has useful fields like email,cognito:username, and cognito:groups (if the user is in any group)
* didn't tested with social providers so far.
* This was working with angular 17.0.7/17.2.2 and aws-amplify 6.0.8/6.0.17

**Comment by nadetastic** [comment]
@myonara glad that the hints from @cwomack have you working again, let us know if you have any more questions on this.

**Comment by cwomack** [comment]
@myonara, glad to hear you're unblocked! I'll close this issue then for now, but feel free to reply back if you still experience this or feel free to open a new issue if there's anything else that you run in to.
